core/trino-spi/src/main/java/io/trino/spi/block/LazyBlock.java

```diff
@@ -362,6 +362,10 @@ public Block getTopLevelBlock()
 
         public Block getFullyLoadedBlock()
         {
+            if (block != null) {
```

Comment:
The commit message is wrong. You are changing `getFullyLoadedBlock` not `getLoadedBlock`